### PR TITLE
Add pagination

### PR DESCRIPTION
**Pull request by nhunzaker**
This commit adds pagination to ArsArsenal. It breaks up the data
fetcher and URL building options to make URL building clearer and
adds logic ot watch the last item in a list to trigger a new page
of data.

I need some way of identicating that pagination is done. In a
follow-up commit, I'd like to add a new option for "Pagination is
done". It would be nice to reduce configuration of this to be bear
minimum.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
   "dependencies": {
     "classnames": "~2.2.6",
     "lodash.sortby": "^4.7.0",
+    "query-string": "^6.2.0",
```
Comment (nhunzaker) [inline package.json]: ArsArsenal will build a query for you. You just have to return an object from the new `listQuery` option

File: src/components/gallery.tsx [hunk 2]
```diff
@@ -17,8 +19,13 @@ interface Props {
   picked: Array<string | number>
 }
 
+function getAnimationDelay(index) {
```
Comment (solomonhawk) [inline src/components/gallery.tsx]: 🔪 ?

File: src/components/table-view/index.tsx [hunk 3]
```diff
@@ -45,15 +46,15 @@ class TableView extends React.Component<Props, State> {
     return Array.isArray(picked) ? picked.indexOf(id) >= 0 : id === picked
   }
 
-  renderRow(item: Record, index: number) {
+  renderRow(item: Record, index: number, list) {
```
Comment (nhunzaker) [inline src/components/table-view/index.tsx]: Ack, this is gonna break TS.

File: src/components/scroll-monitor.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    let nextChild = container.querySelector('[data-scroll="true"]')
+
+    if (!nextChild || nextChild === this.lastChild) {
+      return
+    }
+
+    let outer = container.getBoundingClientRect()
+    let inner = nextChild.getBoundingClientRect()
+
+    if (inner.top < outer.bottom + outer.height * 0.5) {
```
Comment (solomonhawk) [inline src/components/scroll-monitor.tsx]: Is there a simpler and perhaps more performant way to do this using `scrollTop` and `scrollHeight`? The answer may be `no` but I wanted to ask.

File: src/containers/load-collection.tsx [hunk 5]
```diff
@@ -4,33 +4,141 @@
...
+}
+
+class CollectionFetcher extends React.Component<Props, State> {
+  static defaultProps: Props<null> = {
+    ...DEFAULT_OPTIONS,
+    search: '',
+    render: state => null
+  }
+
+  static getDerivedStateFromProps(props: Props, lastState: State) {
```
Comment (solomonhawk) [inline src/containers/load-collection.tsx]: Neat.

**Comment by nhunzaker** [comment]
Types were a wreck. Fixed. I'll send some follow-up PRs to this branch, I still need to add a way to tell the pagination to stop.

**Pull request closed without merging**